bookwyrm/tests/importers/test_isbn_text_import.py

```diff
@@ -0,0 +1,229 @@
+        self.assertEqual(len(retry_items), 1)
+        self.assertEqual(retry_items[0].index, 0)
+        self.assertEqual(retry_items[0].data["ISBN13"], "9781250313195")
+
+    @responses.activate
+    def test_import_data(self):
+        """resolve entry"""
+        job_options = {
+            "include_reviews": False,
+            "privacy": "unlisted",
+        }
+        import_options = {
+            "file_type": "txt",
+            "default_shelf": "to-read",
+        }
+        import_job = self.importer.create_job(
+            self.user, self.csv, job_options=job_options, import_options=import_options
+        )
+        with patch("bookwyrm.preview_images.generate_edition_preview_image_task.delay"):
+            book = models.Edition.objects.create(title="Test Book")
+
+        with patch(
+            "bookwyrm.models.import_job.ImportItem.get_book_from_isbn"
+        ) as resolve:
+            resolve.return_value = book
+            with patch("bookwyrm.importers.importer.handle_imported_book"):
+                import_data(self.importer.service, import_job.id)
+
+        import_item = models.ImportItem.objects.get(job=import_job, index=0)
+        self.assertEqual(import_item.book.id, book.id)
+
+    def test_handle_imported_book(self):
+        """ISBN text import added a book, this adds related connections"""
+        shelf = self.user.shelf_set.filter(identifier="read").first()
+        self.assertIsNone(shelf.books.first())
+
+        import_job = models.ImportJob.objects.create(user=self.user)
+        datafile = pathlib.Path(__file__).parent.joinpath("../data/isbn.txt")
+        csv_file = open(datafile, "r")
+        for index, entry in enumerate(csv_file.read().splitlines()):
+            entry = self.importer.parse_fields(entry, default_shelf="read")
+            import_item = models.ImportItem.objects.create(
+                job_id=import_job.id, index=index, data=entry, book=self.book
+            )
+            break
+
+        with patch("bookwyrm.preview_images.generate_edition_preview_image_task.delay"):
+            with patch("bookwyrm.models.activitypub_mixin.broadcast_task.delay"):
+                handle_imported_book(
+                    self.importer.service, self.user, import_item, False, "public"
+                )
+
+        shelf.refresh_from_db()
+        self.assertEqual(shelf.books.first(), self.book)
+
+        # # uh oh.
+        # readthrough = models.ReadThrough.objects.get(user=self.user)
+        # self.assertEqual(readthrough.book, self.book)
+        # # I can't remember how to create dates and I don't want to look it up.
+        # self.assertEqual(readthrough.start_date.year, 2020)
+        # self.assertEqual(readthrough.start_date.month, 10)
+        # self.assertEqual(readthrough.start_date.day, 21)
+        # self.assertEqual(readthrough.finish_date.year, 2020)
+        # self.assertEqual(readthrough.finish_date.month, 10)
+        # self.assertEqual(readthrough.finish_date.day, 25)
```

Comment:
Everything about readthrough has been commented in the ISBN test.

@mouse How do you suggest I handle dates (added, started, read)? Should everything be `datetime.now()` (depending on the shelf option, of course: `to-read` would not have Date Started and Date Read, and `reading` would not have Date Read)

Reply:
I think if there aren't any dates, no readthrough should be created?

Reply:
That's what I'm thinking, but I was wondering if dates are necessary for some shelves. E.g. is a start date necessary in order for a book to be shelved in the Currently Reading shelf? (that would explain why these books not added to that shelf… or perhaps it's a problem with the identifier?)

Reply:
ah no, it isn't required. `readthrough` is independent from shelves. I'm not sure what's up with how it's shelving, it might be some logic in `models/import_job.py`?